include/nlohmann/json.hpp

```diff
@@ -910,7 +910,10 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
         bool is_an_object = std::all_of(init.begin(), init.end(),
                                         [](const detail::json_ref<basic_json>& element_ref)
         {
-            return element_ref->is_array() && element_ref->size() == 2 && (*element_ref)[0].is_string();
+            // The cast is to ensure op[size_type] is called, bearing in mind size_type may not be int;
+            // (many string types can be constructed from 0 via its null-pointer guise, so we get a
+            // broken call to op[key_type], the wrong semantics and a 4804 warning on Windows)
+            return element_ref->is_array() && element_ref->size() == 2 && (*element_ref)[(size_type)0].is_string();
```

Comment:
I'd recommend `size_type(0)` or `static_cast<size_type>(0)` rather than a c-style cast.

Reply:
I agree - please change this.

Reply:
Roger that.

Reply:
'tis done